org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/IndexAwareNameEnvironment.xtend

```diff
@@ -31,7 +32,15 @@ import org.eclipse.emf.ecore.resource.Resource
 		return findType(className)
 	}
 	
-	def findType(QualifiedName className) {
+	def NameEnvironmentAnswer findType(QualifiedName className) {
+		if (classFileCache.containsKey(className)) {
+			val t = classFileCache.get(className)
+			// TODO is this ok?
+			if (t===null) {
```

Comment:
ConcurrentHashMap cannot contain null. We may need a dedicated null value

Reply:
hmmm we dont store null there nor is the reader place the same as the writer so i am not sure if we need null at all